packages/astro/test/units/integrations/api.test.js

```diff
@@ -131,6 +131,158 @@ describe('Integration API', () => {
+				'/src/pages/about.astro': '',
+				'/src/actions.ts': 'export const server = {}',
+				'/src/foo.astro': '',
+			});
+
+			await runInContainer(
+				{
+					inlineConfig: {
+						root: fixture.path,
+						integrations: [
+							{
+								name: 'test',
+								hooks: {
+									'astro:config:setup': (params) => {
+										params.injectRoute({
+											entrypoint: './src/foo.astro',
+											pattern: '/foo',
+										});
+									},
+									'astro:routes:resolved': (params) => {
+										routes = params.routes.map((r) => ({
+											isPrerendered: r.isPrerendered,
+											entrypoint: r.entrypoint,
+											pattern: r.pattern,
+											params: r.params,
+											origin: r.origin,
+										}));
+										routes.sort((a, b) => a.pattern.localeCompare(b.pattern));
+									},
+								},
+							},
+						],
+					},
+				},
+				async (container) => {
+					assert.deepEqual(
+						routes,
+						[
+							{
+								isPrerendered: false,
+								entrypoint: '../../../../dist/actions/runtime/route.js',
+								pattern: '/_actions/[...path]',
+								params: ['...path'],
+								origin: 'internal',
+							},
+							{
+								isPrerendered: true,
+								entrypoint: 'src/pages/about.astro',
+								pattern: '/about',
+								params: [],
+								origin: 'project',
+							},
+							{
+								isPrerendered: true,
+								entrypoint: 'src/foo.astro',
+								pattern: '/foo',
+								params: [],
+								origin: 'external',
+							},
+						].sort((a, b) => a.pattern.localeCompare(b.pattern)),
+					);
+
+					await fixture.writeFile('/src/pages/bar.astro', '');
+					container.viteServer.watcher.emit(
+						'add',
+						fixture.getPath('/src/pages/bar.astro').replace(/\\/g, '/'),
+					);
+					await new Promise((r) => setTimeout(r, 100));
+
+					assert.deepEqual(
+						routes,
+						[
+							{
+								isPrerendered: false,
+								entrypoint: '../../../../dist/actions/runtime/route.js',
+								pattern: '/_actions/[...path]',
+								params: ['...path'],
+								origin: 'internal',
+							},
+							{
+								isPrerendered: true,
+								entrypoint: 'src/pages/about.astro',
+								pattern: '/about',
+								params: [],
+								origin: 'project',
+							},
+							{
+								isPrerendered: true,
+								entrypoint: 'src/pages/bar.astro',
+								pattern: '/bar',
+								params: [],
+								origin: 'project',
+							},
+							{
+								isPrerendered: true,
+								entrypoint: 'src/foo.astro',
+								pattern: '/foo',
+								params: [],
+								origin: 'external',
+							},
+						].sort((a, b) => a.pattern.localeCompare(b.pattern)),
+					);
+
+					await fixture.writeFile('/src/pages/about.astro', '---\nexport const prerender=false\n');
+					container.viteServer.watcher.emit(
+						'change',
+						fixture.getPath('/src/pages/about.astro').replace(/\\/g, '/'),
+					);
+					await new Promise((r) => setTimeout(r, 100));
+
+					assert.deepEqual(
```

Comment:
This is failing but I can't figure out why. Looks like the partial update of the manifest is not working as expected